### PR TITLE
Move FABs base Security Manager into Airflow.

### PR DESCRIPTION
**Pull request by jhtimmins**
As one of the first steps in refactoring Airflow's usage of the FAB security manager, copy FAB's base SecurityManager classes into Airflow.

### CODE REVIEW

**Comment by uranusjr** [comment]
Do we want to keep FAB in the name? If we end up removing Rlask-Appbuilder altogether, the name reference may feel confusing to newcomers.

**Comment by jhtimmins** [comment]
> Do we want to keep FAB in the name? If we end up removing Flask-Appbuilder altogether, the name reference may feel confusing to newcomers.

I'm refactoring it in pieces, so I plan to change the naming in a PR that's coming shortly. For now though, it exactly matches the FAB architecture and references FAB models/concepts.

**Comment by uranusjr** [comment]
I see, thanks for the clarification.

File: airflow/www/fab_security/manager.py [hunk 1]
```diff
@@ -0,0 +1,1952 @@
+import base64
```
Comment (kaxil) [inline airflow/www/fab_security/manager.py:22]: Need to add Apache-license same as below:

https://github.com/apache/airflow/blob/d4a056647e0f4bf76d1139853917149b907beb2d/airflow/config_templates/default_webserver_config.py#L1-L17

Comment (jhtimmins) [reply airflow/www/fab_security/manager.py:22]: Thanks Kaxil, I've added the license.

File: airflow/www/security.py [hunk 2]
```diff
@@ -32,6 +31,7 @@
 from airflow.security import permissions
 from airflow.utils.log.logging_mixin import LoggingMixin
 from airflow.utils.session import provide_session
+from airflow.www.fab_security.sqla.manager import SecurityManager
```
Comment (ashb) [inline airflow/www/security.py:34]: Is the plan to customize and tweak the content of these files?

If not, It maybe makes more sens to put them under airflow/_vendor/

Or maybe even if we do plan on doing that, if the content is "as upstream" now, maybe it still makes sense to put in that location until we do make such a change.

File: NOTICE [hunk 3]
```diff
@@ -12,3 +12,11 @@ This product contains a modified portion of 'Hue' developed by Cloudera, Inc.
 (https://github.com/cloudera/hue/).
 
 * Copyright 2009-2017 Cloudera Inc.
+
+
+Flask App Builder:
+-----
```
Comment (kaxil) [inline NOTICE:18]:
```suggestion
Flask App Builder:
------------------
```

**Review by ashb — APPROVED** [review]

**Pull request merged**